Configure reports Boost not found when g++ is not available

On CentOS 6 x64:

I have boost-devel (v1.41.0) installed, but not g++.

The configure script failed to detect g++, but attempted to use g++ to see if boost existed:

<pre>checking whether we are using the GNU C++ compiler... no
checking whether g++ accepts -g... no
checking dependency style of g++... none
checking whether we are using the GNU C++ compiler... (cached) no
checking whether g++ accepts -g... (cached) no
checking dependency style of g++... (cached) none
checking whether gcc and cc understand -c and -o together... yes
checking for bison... no
checking for byacc... no
checking for flex... no
checking for lex... no
checking for a BSD-compatible install... /usr/bin/install -c
checking whether make sets $(MAKE)... (cached) yes
checking for ragel... no
checking for asciidoc... no
checking if with lua... checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
no
checking for Boost headers version >= 1.35.0... no
configure: error: cannot find Boost headers version >= 1.35.0
</pre>


Which leads to a very confusing error message, which indicates boost isn't available, when in reality it's just the detection script failing:

config.log shows:

<pre>
configure: this is boost.m4 serial 18
configure:18252: checking for Boost headers version >= 1.35.0
configure:18333: g++ -c -Wall -O2  conftest.cpp >&5
./configure: line 18335: g++: command not found
configure:18340: $? = 127
configure: failed program was:
| /* confdefs.h.  */
| #define PACKAGE_NAME "pdns"
| #define PACKAGE_TARNAME "pdns"
| #define PACKAGE_VERSION "git"
| #define PACKAGE_STRING "pdns git"
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE "pdns"
| #define VERSION "git"
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_DLFCN_H 1
| #define LT_OBJDIR ".libs/"
| /* end confdefs.h.  */
| #include <boost/version.hpp>
| #if !defined BOOST_VERSION
| # error BOOST_VERSION is not defined
| #elif BOOST_VERSION < 103500
| # error Boost headers version < 103500
| #endif
|
| int
| main ()
| {
|
|   ;
|   return 0;
| }
configure:18333: g++ -c -Wall -O2  -I/usr/include conftest.cpp >&5
./configure: line 18335: g++: command not found
configure:18340: $? = 127
configure: failed program was:
| /* confdefs.h.  */
| #define PACKAGE_NAME "pdns"
| #define PACKAGE_TARNAME "pdns"
| #define PACKAGE_VERSION "git"
| #define PACKAGE_STRING "pdns git"
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE "pdns"
| #define VERSION "git"
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_DLFCN_H 1
| #define LT_OBJDIR ".libs/"
| /* end confdefs.h.  */
| #include <boost/version.hpp>
| #if !defined BOOST_VERSION
| # error BOOST_VERSION is not defined
| #elif BOOST_VERSION < 103500
| # error Boost headers version < 103500
| #endif
|
| int
| main ()
| {
|
|   ;
|   return 0;
| }
configure:18369: result: no
configure:18374: error: cannot find Boost headers version >= 1.35.0
</pre>


I've found several other posts by CentOS users on the mailing list complaining of this same error.  They were likely all caused by this.

--- a/configure.ac
+++ b/configure.ac
@@ -18,12 +18,16 @@ AC_C_BIGENDIAN
 : ${CXXFLAGS="-Wall -O2"}
 
 AC_PROG_CC
-AC_PROG_CXX
 AM_PROG_CC_C_O
 AC_PROG_YACC
 AM_PROG_LEX
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
+
+AC_PROG_CXX
+AS_IF([test "x$CXX" = "xno" || test "x$CXX:x$GXX" = "xg++:x"],
+  AC_MSG_ERROR([no C++ compiler found])
+)
 
 AC_LANG([C++])
 

